packages/longhorn/overlay/templates/userroles.yaml

```diff
@@ -0,0 +1,11 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: "longhorn-view"
+  labels:
+    rbac.authorization.k8s.io/aggregate-to-view: "true"
+rules:
+  - apiGroups: [ "longhorn.io" ]
+    resources: [ "volumes", "volumes/status", "engines", "engines/status", "replicas", "replicas/status", "settings",
+                 "engineimages", "engineimages/status", "nodes", "nodes/status", "instancemanagers", "instancemanagers/status" ]
+    verbs: [ "get", "list", "watch" ]
```

Comment:
Can we also get the edit and admin roles added to this same file. This view one looks good. The only real difference between edit and admin is that admin can modify roles/role-bindings, so since this chart doesn't need to do that they should be the same. 

Just as an overview, the goal here is to give a user with the relevant ClusterRole like `edit` all the edit permissions they would need to configure the resources around the chart. And that happens automatically via the k8s ClusterRole aggregation through the label.

Reply:
Oh, I see your other comment that you added the admin/edit in original ClusterRole. We'd prefer to have these clusterRoles only include what is necessary for the user to interact with the chart/longhorn. Not everything needed for the operator to run. So ideally they only include the longhorn CRDs.

Reply:
How about some specifical k8s defined resources like `coordination.k8s.io/lease`? It's not included  in those ClusterRole by default.

Reply:
Would the user need to configure the lease to configure longhorn ? Or does longhorn take care of the lease for them ? 

If they don't need access to it then leave it out. If they do need access to it then we should talk about it. We have another case where we created a separate Role (not a ClusterRole) to specify non-chart resources. That way users don't accidentally assign out ClusterRole with ClusterRoleBinding and give cluster wide access to basic resources.

Reply:
Longhorn itself will need to configure the lease during the launching. Then it seems that I don't need to create one more Role for non-chart resources.